tasks/synopsys-detect-task/ts/DetectScriptDownloader.ts

```diff
@@ -7,20 +7,27 @@ import fileSystem, {WriteStream} from 'fs';
 import {PathResolver} from './PathResolver';
 
 export class DetectScriptDownloader {
-    static readonly DETECT_DOWNLOAD_URL = 'https://detect.synopsys.com'
+    static readonly DETECT_DOWNLOAD_URL = 'https://detect.blackduck.com'
+    static readonly DETECT_DOWNLOAD_FALLBACK_URL = 'https://detect.synopsys.com'
 
     private constructor() {}
 
-    static async downloadScript(proxyInfo: IProxyInfo | undefined, scriptName: string, scriptDirectory: string): Promise<boolean> {
-        logger.info('Downloading detect script.')
+    static async downloadScript(proxyInfo: IProxyInfo | undefined, scriptName: string, scriptDirectory: string, isBlackDuckAccessible: boolean): Promise<boolean> {
         if (!fileSystem.existsSync(scriptDirectory)) {
             fileSystem.mkdirSync(scriptDirectory, {recursive: true})
         }
 
-        const downloadLink: string = this.getFullDownloadUrl(scriptName)
+        let downloadLink:string
+        if(isBlackDuckAccessible) {
+            downloadLink = this.getFullDownloadUrl(scriptName)
+        } else {
+            downloadLink = this.getFullFallbackDownloadUrl(scriptName)
+        }
```

Comment:
The logic on lines 20 to 25 could potentially be pushed down into the getFullDownloadUrl function.